onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -650,6 +725,38 @@ void InitializeSession(InferenceSession* sess, const std::vector<std::string>& p
   OrtPybindThrowIfError(sess->Initialize());
 }
 
+bool CheckIfTensor(const InputDefList& def_list,
+                   const std::string& name,
```

Comment:
nit: Maybe CheckInputIsTensor given it's also testing if it's a valid input name.

Also can some of the helper methods here be static?

Reply:
Actually we invoke this check while binding inputs/outputs. Maybe I should just rename the parameters ?

Reply:
Seems like we have 2 typedefs - InputDefList and OutputDefList which are both a vector of NodeArg pointers. Maybe we can have one more - DefList (which is generic) and we can scratch the Input whichmakes the function misleading to the code reader.

Reply:
or just don't use a typedef here and take std::vector<const onnxruntime::NodeArg*>& as input.

Reply:
Done

Reply:
all helpers are static now